core/src/main/java/org/elasticsearch/http/netty/NettyHttpServerTransport.java

```diff
@@ -312,6 +324,41 @@ protected void doStart() {
         this.boundAddress = new BoundTransportAddress(boundAddresses.toArray(new TransportAddress[boundAddresses.size()]), new InetSocketTransportAddress(publishAddress));
     }
 
+    private CorsConfig buildCorsConfig(Settings settings) {
+        if (!settings.getAsBoolean(SETTING_CORS_ENABLED, false)) {
+            return CorsConfigBuilder.forOrigins().disable().build();
+        }
+        String origin = settings.get(SETTING_CORS_ALLOW_ORIGIN);
+        final CorsConfigBuilder builder;
+        if (Strings.isNullOrEmpty(origin)) {
+            builder = CorsConfigBuilder.forOrigins();
+        } else if (origin.equals(ANY_ORIGIN)) {
+            builder = CorsConfigBuilder.forAnyOrigin();
+        } else {
+            Pattern p = RestUtils.checkCorsSettingForRegex(origin);
+            if (p == null) {
+                builder = CorsConfigBuilder.forOrigins(RestUtils.corsSettingAsArray(origin));
+            } else {
+                builder = CorsConfigBuilder.forPattern(p);
+            }
+        }
+        if (settings.getAsBoolean(SETTING_CORS_ALLOW_CREDENTIALS, false)) {
+            builder.allowCredentials();
+        }
+        String[] strMethods = settings.getAsArray(SETTING_CORS_ALLOW_METHODS, new String[0]);
+        HttpMethod[] methods = Arrays.asList(strMethods)
+                                     .stream()
+                                     .map(m -> HttpUtils.toHttpMethod(m))
+                                     .filter(Optional::isPresent)
+                                     .map(Optional::get)
+                                     .toArray(size -> new HttpMethod[size]);
```

Comment:
looks more tedious than just filling an array, but I didnt do an awful lot of java 8 stuff yet :-)